packages/service-provider-core/src/printable-bson.spec.ts

```diff
@@ -95,6 +95,34 @@ describe('BSON printers', function () {
     ).to.equal("MD5('0123456789abcdef0123456789abcdef')");
   });
 
+  describe('with Vector types', function () {
+    it('formats Int8Array correctly', function () {
+      expect(
+        inspect(bson.Binary.fromInt8Array(new Int8Array([1, 2, 3])))
+      ).to.equal('Binary.fromInt8Array(new Int8Array([ 1, 2, 3 ]))');
+    });
+
+    it('formats PackedBits correctly', function () {
+      expect(
+        inspect(bson.Binary.fromPackedBits(new Uint8Array([1, 2, 3])))
+      ).to.equal('Binary.fromPackedBits(new Uint8Array([ 1, 2, 3 ]))');
+    });
+
+    it('formats Float32Array correctly', function () {
+      expect(
+        inspect(
+          bson.Binary.fromFloat32Array(
+            new Float32Array([1.1111, 2.2222, 3.3333])
+          ),
+          { compact: true }
+        )
+      ).matches(
+        // Precision is lost because of float handling, so we use regex to match
+        /Binary.fromFloat32Array\(new Float32Array\(\[ 1\.1\d*, 2\.2\d*, 3\.3\d* \]\)\)/
```

Comment:
not sure if we're meant to be dealing with float precision weirdness in JS in this case. can definitely see how it'd be weird to see imprecise floats while trying to inspect data. we could do something like https://github.com/mongodb-js/compass/pull/6834/files#diff-3d4b298fbb2cfe42467edc59277483eb9bcfdc46b07ea658ab3dc4f66903da88R174